### PR TITLE
Add HttpClient tests to support all HTTP verbs

### PR DESCRIPTION
**Pull request by davidsh**
Added tests for validating against all standard and custom HTTP verbs. These tests use the new networking test server.

Cosolidated some existing tests that were duplicative. More test porting and reorganzing will be done in later PRs.

### CODE REVIEW

**Comment by davidsh** [comment]
@stephentoub @CIPop PTAL
/cc: @vijaykota @kapilash

**Comment by davidsh** [comment]
Test failures on Linux:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_prtest/5931/

**Comment by davidsh** [comment]
Looks a problem with parsing HTTP response headers:

>
MESSAGE:
Assert.Equal() Failure ↓ (pos 0) Expected: GET Actual: GET ↑ (pos 0)
+++++++++++++++++++
STACK TRACE:
at System.Net.Http.Functional.Tests.TestHelper.VerifyRequestMethod(HttpResponseMessage response, String expectedMethod) at System.Net.Http.Functional.Tests.HttpClientHandlerTest.<SendAsync_SendRequestUsingMethodToEchoServerWithContent_Success>d__117.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)

**Comment by stephentoub** [comment]
> Looks like a problem with parsing

The issues all boil down to a space at the beginning of what's being compared:
```
16:18:19       Expected: GET
16:18:19       Actual:    GET
```

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 1]
```diff
@@ -33,33 +34,54 @@ public class HttpClientHandlerTest
...
+                    new object[] { "POST", true },
+                    new object[] { "PUT", false },
+                    new object[] { "PUT", true },
+                    new object[] { "DELETE", false },
+                    new object[] { "DELETE", true },
+                    
+                    // Custom HTTP method
+                    new object[] { "CUSTOM1", false },
+                    new object[] { "CUSTOM1", true }
+                };
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Consider instead something like:
```C#
public readonly static IEnumerable<object[]> HttpMethods = GetMethods("GET", "HEAD", "POST", "PUT", "DELETE", "OPTIONS", "TRACE", "CUSTOM1");
public readonly static IEnumerable<object[]> HttpMethodsThatAllowContent = GetMethods("GET", "POST", PUT", "DELETE", "CUSTOM1");

private static IEnumerable<object[]> GetMethods(params string[] methods)
{
    foreach (string method in methods)
    foreach (bool secure in new[] { true, false })
        yield return new object[] { method, secure };
}
```

**Comment by stephentoub** [comment]
LGTM

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: @stephentoub Thx. I knew there had to be a better way.  You're the `yield` guru!

Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: :smile:

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 2]
```diff
@@ -24,6 +24,7 @@ namespace System.Net.Http.Functional.Tests
     public class HttpClientHandlerTest
     {
         readonly ITestOutputHelper _output;
+        private const string ExpectedContent = "Test contest";
```
Comment (vijaykota) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Nit: I think you meant "Test conte<b><u>n</u></b>t"

Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: >Nit: I think you meant "Test content"

Thx. We all missed that. I will fix it later in the week when I do my next set of test changes.

**Comment by vijaykota** [comment]
LGTM. Thanks for refactoring, @davidsh

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Maybe this was just David's way of challenging the code to get it right, a "contest" if you will :wink:

**Pull request merged**